include/boost/uuid/detail/random_provider_wincrypt.ipp

```diff
@@ -0,0 +1,86 @@
+/* boost uuid/detail/random_provider_wincrypt implementation
+*
+* Copyright Jens Maurer 2000
+* Copyright 2007 Andy Tompkins.
+* Copyright Steven Watanabe 2010-2011
+* Copyright 2017 James E. King, III
+*
+* Distributed under the Boost Software License, Version 1.0. (See
+* accompanying file LICENSE_1_0.txt or copy at
+* http://www.boost.org/LICENSE_1_0.txt)
+*
+* $Id$
+*/
+
+#include <boost/core/ignore_unused.hpp>
+#include <boost/winapi/crypt.hpp>
+#include <boost/winapi/get_last_error.hpp>
+#include <boost/throw_exception.hpp>
+#include <string>
+
+#if !defined(BOOST_UUID_RANDOM_PROVIDER_NO_LIB)
+#   if defined(_WIN32_WCE)
+#      define BOOST_LIB_NAME "coredll"
+#   else
+#      define BOOST_LIB_NAME "advapi32"
+#   endif
+#   define BOOST_AUTO_LINK_NOMANGLE
+#   include <boost/config/auto_link.hpp>
+#   undef BOOST_AUTO_LINK_NOMANGLE
+#endif
+
+namespace boost {
+namespace uuids {
+namespace detail {
+
+class random_provider_base
+{
+  public:
+    random_provider_base()
+        : hProv_(NULL)
+    {
+        if (!
+#if defined(BOOST_NO_ANSI_APIS)
+          boost::winapi::CryptAcquireContextW
+#else
+          boost::winapi::CryptAcquireContextA
```

Comment:
No reason to ever use the A function here, if you ask me.

Reply:
I believe you may be right now that the provider string / enumeration stuff is gone.  If so I'll simplify and remove the additional test for this macro.